lib/iris/_shapefiles.py

```diff
@@ -3,241 +3,447 @@
+    ----------
+    geometry : :class:`shapely.Geometry`
+        The geometry to transform.
+    geometry_crs : :class:`cartopy.crs`, :class:`pyproj.CRS`
+        The coordinate reference system of the geometry.
+    cube_crs : :class:`cartopy.crs`, :class:`pyproj.CRS`
+        The coordinate reference system of the cube.
 
-def _trans_func(geometry):
-    """Pocket function for transforming the x coord of a geometry from -180 to 180 to 0-360."""
-    for point in geometry:
-        if point[0] < 0:
-            point[0] = 360 - np.abs(point[0])
-    return geometry
+    Returns
+    -------
+    :class:`shapely.Geometry`
+        The transformed geometry.
+    """
+    # Set-up transform via pyproj
+    t = Transformer.from_crs(
+        crs_from=geometry_crs, crs_to=cube_crs, always_xy=True
+    ).transform
+    # Transform geometry
+    return shapely.ops.transform(t, geometry)
 
 
-def _cube_primary_xy_coord_names(cube):
-    """Return the primary latitude and longitude coordinate names, or long names, from a cube.
+def _get_weighted_mask(
+    cube: iris.cube.Cube,
+    geometry: shapely.Geometry,
+    minimum_weight: float,
+) -> np.array:
+    """Get a mask based on the geometry and minimum weight.
 
     Parameters
     ----------
     cube : :class:`iris.cube.Cube`
+        The cube to mask.
+    geometry : :class:`shapely.Geometry`
+        The geometry to use for masking.
+    minimum_weight : float
+        The minimum weight of the geometry to be included in the mask.
+        If the weight is less than this value, the geometry will not be
+        included in the mask.
 
     Returns
     -------
-    tuple of str
-        The names of the primary latitude and longitude coordinates.
-
+    :class:`np.array`
+        An array of the shape of the x & y coordinates of the cube, with points
+        to mask equal to True.
     """
-    latc = (
-        cube.coords(axis="y", dim_coords=True)[0]
-        if cube.coords(axis="y", dim_coords=True)
-        else -1
-    )
-    lonc = (
-        cube.coords(axis="x", dim_coords=True)[0]
-        if cube.coords(axis="x", dim_coords=True)
-        else -1
-    )
-
-    if -1 in (latc, lonc):
-        msg = "Error retrieving 1d xy coordinates in cube: {!r}"
-        raise ValueError(msg.format(cube))
+    if not cube.coord(axis="X", dim_coords=True).has_bounds():
+        cube.coord(axis="X", dim_coords=True).guess_bounds()
+    if not cube.coord(axis="Y", dim_coords=True).has_bounds():
+        cube.coord(axis="Y", dim_coords=True).guess_bounds()
+    # Get the shape of the cube
+    w = len(cube.coord(axis="X", dim_coords=True).points)
+    h = len(cube.coord(axis="Y", dim_coords=True).points)
+    # Get the bounds of the cube
+    # x_bounds = _get_mod_rebased_coord_bounds(cube.coord(x_name))
+    # y_bounds = _get_mod_rebased_coord_bounds(cube.coord(y_name))
+    x_bounds = cube.coord(axis="X", dim_coords=True).bounds
+    y_bounds = cube.coord(axis="Y", dim_coords=True).bounds
```

Comment:
Please just assign the X and Y coords to dedicated variables, this is very difficult to read (and slow to run)! 😅